server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -566,12 +589,14 @@ private static final class ThreadContextStruct {
             Collections.emptyMap(),
             Collections.emptyMap(),
             Collections.emptyMap(),
+            Collections.emptyMap(),
             false
         );
 
         private final Map<String, String> requestHeaders;
         private final Map<String, Object> transientHeaders;
         private final Map<String, Set<String>> responseHeaders;
+        private final Map<String, Object> persistentHeaders;
```

Comment:
Setting aside bad actors, the underlying data structure is a mutable map, shouldn't this be a add only map to further limit misuse / tampering?

Reply:
This data structure is private and any publicly exposed method on ThreadContext struct uses `putSingleHeader` when putting the header and value into the map. If the header is always present an IllegalArgumentException is thrown: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java#L626-L630

```
private static <T> void putSingleHeader(String key, T value, Map<String, T> newHeaders) {
    if (newHeaders.putIfAbsent(key, value) != null) {
        throw new IllegalArgumentException("value for key [" + key + "] already present");
    }
}
```